### ISSUE TITLE
no event activity in catalog even if pixel installed

### ISSUE THREAD

**Issue by kg0925** [facebookincubator/Facebook-For-OpenCart#24]
recently i installed this module to my opencart store 3.0.2.0 . it installed successfully. pixel is also installed correctly. i checked product page, category page, add to cart, initial checkout, purchase. pixel show correctly. but it's not showing any activity on catalog -&gt; event data sources. it shows no events. if i want to create dynamic ads then in catalog it should be shown. but there is nothing. it doesn't show any events there. i checked pixel by demo purchase on website but it shows no Events Matched to Products in Your Catalogs

website : fashiondeal.in

**Comment by shue249** [comment]
Thank you for your feedback.

We went to your website but did not found the pixel installed. Did you delete away the Facebook Ads Extension settings? If you like us to continue to check, will appreciate if you can send us your pixel id.

Some of the event activities may take some time, ~24 hours, to be reflected on the dashboards (eg catalog or pixel).

**Comment by kg0925** [comment]
i had uninstalled the module. now i installed it again. please check now and also let me know how to remove that cookie policy tab from the website bottom. i don't to be shown it to visitors.

**Comment by shue249** [comment]
@kg0925 We can see that the plugin is installed correctly and the events are fired correctly on the product catalog event source (see the ViewContent and AddToCart). Regarding your earlier concern on the purchase warning, this is because we have yet to receive Purchase events. Once consumers start to make purchases on your website, the dashboard will show out the purchase events.

With regards to the cookie policy tab, this is a requirement for GDPR and is only shown to consumers from countries which requires GDPR (mainly EU countries). You can read up more details on GDPR here - https://www.facebook.com/business/gdpr.

As the installation is successful, we will proceed to close off this ticket, but feel free to open up the ticket if you have any further questions.

**Comment by kg0925** [comment]
i understand that it's part of facebook but it can be hide right?? with
manual pixel installation there is no such tab so why it can't be with this
plugin. i don't want to be shown it on my website to affect user
experience. please help me to remove it.

**Comment by kg0925** [comment]
i removed that cookie tab code and reinstalled the plugin now pixel is not
showing in pixel helper.

**Comment by kg0925** [comment]
it shows this error   client denied by server configuration:
/home/myusername/public_html/ads.txt

**Comment by shue249** [comment]
What do you see when you go into Facebook Ads Extension from the OpenCart admin panel? Do you see the setup completed with a Store ID? If do not see the store ID, can you setup the Facebook Ads Extension again?

**Comment by kg0925** [comment]
yes it shows store id everything same as before. i also tried settings and
setup again. but still no luck. pixel stop showing after removing that
cookie tab code.

**Comment by kg0925** [comment]
i just started setup 15 minutes ago currently extension uploading the
catalog to facebook. can you check it now is it showing pixel to you?
url - fashiondeal.in
for me it's not showing pixel installed since i removed that cookie code.

**Comment by kg0925** [comment]
i tried again by uninstall previous one installed this one. but still pixel
doesn't show in pixel helper. now even if i installed the fresh plugin with
the cookie tab. it also not working for pixel. seems like pixel is
completely disabled for this plugin.

**Comment by shue249** [comment]
Can I check if you have click on the Refresh button in the Modifications panel? This step is required for the plugin to take effect after every new installation.

![image](https://user-images.githubusercontent.com/11957585/44062379-a4e57326-9f8e-11e8-95e2-9f011ddb7a06.png)

**Comment by kg0925** [comment]
i already did that. after each and every installation i do refresh cache in
modification.

**Comment by shue249** [comment]
@kg0925 We went to your website and it is showing as under maintenance.

At the same time, I want to share with you on how we check if the plugin is installed correctly.
![image](https://user-images.githubusercontent.com/11957585/44072020-14ce36ac-9fbf-11e8-9154-cd7b2462fef0.png)

Can I understand if you have deleted away the Facebook Ads Extension settings? I can suggest you to retain the settings and not to delete them away. That will make the debugging faster from our end too.

**Comment by kg0925** [comment]
i figured it out. journal 2 theme is blocking pixel code. in default theme
pixel shows. so i am reinstalling the journal theme.

**Comment by shue249** [comment]
@kg0925 

1. We tested on our test servers and our plugin is still firing the pixels on journal theme

2. I remembered that earlier, you managed to get the plugin setup and the pixels are firing. See the 4th post above where I did check your website and found the pixels fired. Are there any other 3rd party plugins?

3. Which OpenCart version are you on?

![image](https://user-images.githubusercontent.com/11957585/44087090-36ffe784-9ff1-11e8-9a64-4cd56ec8c7be.png)

**Comment by kg0925** [comment]
My store is on opencart 3.0.2.0 version. Yes, i have installed many third
party modules. If other module is causing this problem. Can you suggest how
to find that module.

**Comment by shue249** [comment]
@kg0925 I am wondering if you have made any changes to your OpenCart setup? We went in to check and found that the pixels are firing correctly and the plugin version does not contain the cookie bar.

Also, do take note that if you will to upgrade the plugin to newer versions subsequently, you will need to do the same changes to remove the cookie bar otherwise the cookie bar will still continue to be shown.

Will proceed to close off this ticket.

**Comment by kg0925** [comment]
even i am wondering pixels are really firing???  but i am not able to see
it firing. from my side pixel helper is not showing anything. i just
reinstalled the plugin after removing that cookie tab code.

**Comment by shue249** [comment]
@kg0925 Here is a screenshot of your website from my computer showing the pixels fired. Are you using any ad blockers?

![image](https://user-images.githubusercontent.com/11957585/44101892-03d7dad6-a01b-11e8-8732-ddb4baf37792.png)

**Comment by kg0925** [comment]
yes i am using adblocker in chrome but tried after disabling it. but still
pixel is not showing.

**Comment by kg0925** [comment]
also when i check event manager in Facebook business manager. event pixels
is not firing properly. it shows pageview event last received 51 minutes
ago. but my website is visited by above 150 visitors in this time frame.

**Comment by shue249** [comment]
Will you be able to check on View Source of your website if you can find the pixel ID on the page? 

The information on pixel dashboard/events manager may be slightly delayed. For more real time updates, you can check on Facebook Analytics - https://www.facebook.com/analytics. I can see that you have some ViewContent and AddToCart events.

![image](https://user-images.githubusercontent.com/11957585/44125886-9ba8061e-a067-11e8-8463-9b7bbbddd32f.png)

**Comment by kg0925** [comment]
yes in view source of my website pixel id is shown. i think that pixel code
script is blocking randomly somewhere. i looked in to analytics it shows
only 6 visits yesterday. so it's confirmed that there is something wrong.
previously it was working great all pixel was firing with third party
plugins.there was no issue there. i just wanted to remove that cookie bar
after that this facebook plugin stucked.

this issue start happening when i uninstalled the plugin with cookie bar,
after i remove that cookie tab code in  install.xml from the downloaded
plugin. after i installed that modified plugin. then cleared cache from
dashboard developer setting. after i cleared journal 2 theme cache. and i
visited the website and found pixel is disappeared. till that time i am
facing this issue. even if i uninstalled that opencart and installed fresh
version and restored my data and installed this module still it's not
working. is there any possibilities that host server can block any
script??   and yes i am using cloud flare also. i tried disabling cloud
flare also but no result.

**Comment by kg0925** [comment]
and yes i am on php 7  is this plugin supported to php version 7??

**Comment by shue249** [comment]
1. Yes our plugin supports PHP7. It supports PHP 5.4 and above
2. The pixel code script is run on consumer's computer/mobile phones, as it is a website page. As long as you see the pixel_id in view-source, the pixel should be fired.
3. I go back to your website and strangely, the pixel_id is no longer there. Want to confirm if you have delete the Facebook Ads Extension Settings?

**Comment by kg0925** [comment]
yes i uninstalled the plugin. now i am installing pixels manually. plugin
is not working for me so now i must install it manually. pixel code working
fine manually. you can check now.

**Comment by kg0925** [comment]
hello are you sure is it compatible with php 7. i tried this plugin to
another server with same third party plugins which i am using. your module
is working there and php version is 5.6 to that server.

**Comment by shue249** [comment]
Thank you for your feedback. 

Yes, our plugin supports php7, as our test servers are on php 7.

Can I also understand if you have vqmod installed?

![image](https://user-images.githubusercontent.com/11957585/44237851-be2c7480-a1e4-11e8-814d-0e58a50a8b68.png)

**Comment by kg0925** [comment]
no vqmod is not installed. i tested everything i switched opencart version
from 3.0.2.0 to   2.3.0.2 fresh installation. no third party plugins
installed and then installed the facebook plugin but it's not working. then
i tested on different server with same settings and data backup and
facebook plugin worked there even if third party plugins installed. so i
don't think other module is blocking pixel. there is something wrong on my
current server settings. it's blocking the script. i already contacted my
hosting providers but they are very slow in response.

**Comment by kg0925** [comment]
hello thanks for your support. now extension is working great. my hosting
provider done something and it starts working.

**Comment by kg0925** [comment]
please check from your side also and let me know.

**Comment by kg0925** [comment]
the problem is coockie tab still appearing even if i removed cookie tab
code from the install.xml

**Comment by shue249** [comment]
@kg0925 I did a check on your website and found that you switched to putting pixels manually? Is there a way for you to install back the plugin (using the version which we provided with no cookie bar) so that we can help you check?

**Comment by kg0925** [comment]
i was getting this error PHP Notice:  Undefined index: filter_name in
/home/fashiond/public_html/catalog/controller/extension/facebookevents.php
on line 338

moudle is not working again.

**Comment by shue249** [comment]
Thank you for the feedback. Let us take a look at this specific problem.

**Comment by kg0925** [comment]
hello i just installed the plugin and i just found this error in error
manager. 2018-09-20 9:58:28 - PHP Notice:  Undefined index: filter_name in
/home/fashiond/public_html/catalog/controller/extension/facebookevents.php
on line 338
